salt/grains/core.py

```diff
@@ -2041,7 +2045,7 @@ def _hw_data(osdata):
         }
         for key, fw_file in sysfs_firmware_info.items():
             contents_file = os.path.join('/sys/class/dmi/id', fw_file)
-            if os.path.exists(contents_file):
+            if _file_readable(contents_file):
```

Comment:
Technically, it would be best to this first:
```if osdata['kernel'] == 'Linux' and os.path.exists('/sys/class/dmi/id'):```
and enrich the collected data with other methods (as tested below) if possible/needed. I think that most of the time the other methods don't help much to collect data in containers.